adserver/views.py

```diff
@@ -766,6 +772,161 @@ def get_success_url(self):
         )
 
 
+class AdvertisementBulkCreateView(
+    AdvertiserAccessMixin,
+    UserPassesTestMixin,
+    FormView,
+):
+    """
+    Bulk create view for advertisements.
+
+    Data is validated on upload and then stored in a signed field while being previewed.
+    When the user submits the previewed data, the signed data is stored in the database.
+    """
+
+    form_class = BulkAdvertisementUploadCSVForm
+    model = Advertisement
+    template_name = "adserver/advertiser/advertisement-bulk-create.html"
+    MAXIMUM_SIGNED_TIME_LENGTH = 24 * 60 * 60
+
+    def dispatch(self, request, *args, **kwargs):
+        self.advertiser = get_object_or_404(
+            Advertiser, slug=self.kwargs["advertiser_slug"]
+        )
+        self.flight = get_object_or_404(
+            Flight,
+            slug=self.kwargs["flight_slug"],
+            campaign__advertiser=self.advertiser,
+        )
+        return super().dispatch(request, *args, **kwargs)
+
+    def post(self, *args, **kwargs):
+        if "advertisements" in self.request.FILES:
+            # This is step 1 of uploading the file
+            form = self.get_form()
+            if form.is_valid():
+                # CSV uploaded successfully - show preview
+                preview_ad_type = self.flight.campaign.allowed_ad_types(
+                    exclude_deprecated=True
+                ).first()
```

Comment:
Do we want to let users set this in the future? Currently I'm assuming you can't upload text-only ads. I guess we could add it to the template in the future.

We probably want to make it explicit tho, like `image-v1`, and not just `first()`?

Reply:
`first()` is not random. It is picked based on the `order` so we can control which format gets chosen. However, what if we enable a larger ad format and that's the only valid ad format for that flight? We want it to pick that one.

Reply:
Yea, I know we can make `first()` return the same thing each time. But presumably we want to just define what we want it to do, instead of using this roundabout way to get an ad type?

Reply:
This is just used to display the preview. It just takes the first ad type (ordered by the `order` field which we control) and uses that to display the preview. That seemed like a reasonable choice that would work here as well as on the few campaigns that use other ad types (more in the future?). Happy to do an alternative if you have a better idea.